spring-ai-core/src/main/java/org/springframework/ai/ResourceUtils.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.springframework.core.io.DefaultResourceLoader;
+
+/**
+ * @author Christian Tzolov
+ */
+public class ResourceUtils {
+
+	public static String getText(String uri) {
```

Comment:
Maybe there should be a constructor to Document that takes a Resource and/or have the current String constructor try to parse the passed in text to see if it is a resource string and then take appropriate action.  I guess that would mean you can't create a document with the test "classpath:..."